### PR TITLE
Feature/mat-1049

### PR DESCRIPTION
**Pull request by adongare**
MAT-1049 Implement ruby mongoid templates to generate mongoid models
- Added handlebar templates for ruby mongoid
- Added generator for mongoid models
- Added few tests for mongoid model generation script

### CODE REVIEW

**Comment by AndrewBird81** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5610987&bid=18382456)

File: test/generateMongoidTypes.test.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+describe('generateMongoidTypes', () => {
+
+  const modelDir = './generated/mongoid';
+  // remove models after all tests have been run
+  afterAll(() => {
+    exec(`rm -rf ${path.resolve(modelDir)}`);
+  });
+
+  interface Result {
```
Comment (AndrewBird81) [inline test/generateMongoidTypes.test.ts:15]: Codacy Issue found: [interface name must start with a capitalized I](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5610987)

File: src/generators/MongoidTypeGenerator.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import FileWriter from "../FileWriter";
+import { mongoidPrimitiveTypes } from "../model/dataTypes/primitiveDataTypes";
+import TypeInfo from "../model/TypeInfo";
+import classTemplate from "../templates/rubymongoid/classTemplate";
+import IGenerator from "./IGenerator";
+
+export default class MongoidTypeGenerator implements IGenerator {
```
Comment (adongare) [inline src/generators/MongoidTypeGenerator.ts:7]: @AndrewBird81 I just copy pasted this from typescript model generator. we can discuss if we want to introduce abstract class or move some common code

File: src/generateMongoidTypes.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env node
```
Comment (AndrewBird81) [inline src/generateMongoidTypes.ts:1]: @adongare  Can you take a stab at refactoring this duplicated code into a parent class?

File: src/generators/MongoidTypeGenerator.ts (hunk 2 above)
Comment (AndrewBird81) [reply src/generators/MongoidTypeGenerator.ts:7]: yeah, go ahead and make an abstract class

File: test/generateMongoidTypes.test.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { exec } from 'child_process';
+import fs from 'fs';
+import { mongoidPrimitiveTypes } from '../src/model/dataTypes/primitiveDataTypes';
+import parser from '../src/parser';
+
+describe('generateMongoidTypes', () => {
+
+  const modelDir = './generated/mongoid';
```
Comment (AndrewBird81) [inline test/generateMongoidTypes.test.ts:8]: let's make this directory more test-specific

File: test/generateMongoidTypes.test.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  // remove models after all tests have been run
+  afterAll(() => {
+    exec(`rm -rf ${modelDir}`);
+  });
+
+  function generateMongoids(): Promise<{ code: number, error: any }> {
+    return new Promise(resolve => {
```
Comment (AndrewBird81) [inline test/generateMongoidTypes.test.ts:16]: I don't think we need to use "exec" here- can't we just call the function normally?

File: test/templates/helpers/templateHelpers.test.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { getMongoidPrimitive, removeNamespace} from '../../../src/templates/helpers/templateHelpers';
+
+describe('templateHelpers', () => {
+  describe("getMongoidPrimitive()", () => {
+    test('Should return equivalent mongoid compatible primitive', () => {
+      expect(getMongoidPrimitive('string')).toEqual('String');
+      expect(getMongoidPrimitive('instant')).toEqual('DateTime');
```
Comment (AndrewBird81) [inline test/templates/helpers/templateHelpers.test.ts:7]: Let's test all of the primitive types

File: src/generateMongoidTypes.ts (hunk 3 above)
Comment (adongare) [reply src/generateMongoidTypes.ts:1]: Done

File: test/generateMongoidTypes.test.ts (hunk 5 above)
Comment (adongare) [reply test/generateMongoidTypes.test.ts:18]: it's a script so I need to execute it the way package.json executes it.

File: test/templates/helpers/templateHelpers.test.ts (hunk 6 above)
Comment (adongare) [reply test/templates/helpers/templateHelpers.test.ts:14]: Done

File: test/generateMongoidTypes.test.ts (hunk 4 above)
Comment (adongare) [reply test/generateMongoidTypes.test.ts]: Done

File: package.json [hunk 7]
```diff
@@ -13,13 +13,15 @@
   "scripts": {
     "prebuild": "rm -rf dist; rm -rf generated",
     "build": "tsc",
-    "generateTypeScript": "ts-node src/generateTypeScript.ts",
+    "generateTypeScriptTypes": "ts-node src/generateTypeScriptTypes.ts --output-directory='./generated/typescript'",
+    "generateMongoidTypes": "ts-node src/generateMongoidTypes.ts --output-directory='./generated/mongoid'",
```
Comment (adongare) [inline package.json:17]: @AndrewBird81 I can combine these two scripts into single. let me know your thoughts.

File: src/generateTypeScriptTypes.ts [hunk 8]
```diff
@@ -1,13 +1,12 @@
 #!/usr/bin/env node
-
-import { program } from "commander";
-import { version } from "../package.json";
+import GeneratorProgram from "./GeneratorProgram";
 import TypeScriptGenerator from "./generators/TypeScriptGenerator";
-import parser from "./parser";
+import { program } from "commander";
```
Comment (AndrewBird81) [inline src/generateTypeScriptTypes.ts:4]: Codacy Issue found: [Import sources within a group must be alphabetized.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5610987)

**Comment by AndrewBird81** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5610987&bid=18382456)

**Pull request merged**